### PR TITLE
Fix(header) icon sizing and spacing

### PR DESCRIPTION
**Pull request by vibdev**
## Summary

This fixes issue #420 where header icons are squished in IE 11. Also turns out this also fixes an issue with the centeredNav prop that was masked by not having a centeredMenu ={true} with no leftSlot story. 

## Checklist

- [x] branch has been rebased on the latest master commit
- [x] tests are changed or added
- [x] `yarn test` passes
- [x] code has been documented and, if applicable, usage described in README.md
- [x] design approved final implementation
- [x] a11y approved final implementation
- [x] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/_labs/header/react/lib/Header.tsx [hunk 1]
```diff
@@ -197,6 +197,7 @@ const ChildrenSlot = styled('div')<Pick<HeaderProps, 'centeredNav' | 'themeColor
     '> *:last-child': {
       marginRight: isCollapsed ? '' : 0,
     },
+    width: !isCollapsed ? '100%' : 'auto',
```
Comment (ginna-baker) [inline modules/_labs/header/react/lib/Header.tsx]: should this be auto or just 'unset'?

**Comment by vibdev** [comment]
Looks like there is an issue with the Full Header stories, nav is centered now

**Comment by lychyi** [comment]
@vibdev Updated the branch again, trying to get these snapshots to pass on the right baselines

@ginna-baker thanks for the review!

File: modules/_labs/header/react/stories/stories.tsx [hunk 2]
```diff
@@ -333,7 +333,7 @@ storiesOf('Labs|Header/React', module)
       <div css={containerStyle}>
         <Header
           variant={Header.Variant.Dub}
-          title="Centered Menu"
+          title="Centered Menu With Search"
```
Comment (ginna-baker) [inline modules/_labs/header/react/stories/stories.tsx]: nb Looks like this added text is what's pushing the Logout button over the boundary on the screenshot https://www.chromaticqa.com/snapshot?appId=5d854c26ba934e0020f5e98a&id=5e2b41d359313d0022640f60

Comment (vibdev) [reply modules/_labs/header/react/stories/stories.tsx]: since the responsive stuff is by prop the smaller screenshots can get squished or overflow

**Comment by vibdev** [comment]
@lychyi Thanks for fixing the snaps. For the chromatic, looks like icons and centering is better now. The only thing that is standing out to me is the first dub header (without no icons or avatar) the search bar is now left aligned. I think that is more correct, but wanted to double check.

**Comment by lychyi** [comment]
@vibdev @ginna-baker

I added an alternative approach in #427 if you want to take a look. I think it fixes it in the spirit that the header and the slots were supposed to work (fixed branding slot, then evenly flexed children). Let me know if the fix makes sense to you and if adding `overflow: hidden` or not affects your use cases.

**Comment by vibdev** [comment]
@lychyi you method seems to work, but still have a question about the first dub header (without no icons or avatar); the search bar is now left aligned. Seems kinda weird that the left slot prop is on the right in just that one case. RE: overflow: hidden that breaks search and avatar dropdowns.

File: modules/_labs/header/react/stories/stories.tsx (hunk 2 above)
Comment (NicholasBoll) [reply modules/_labs/header/react/stories/stories.tsx]: IE11 handles this lack of space differently. IE11 is squishing the navigation tabs while Chrome and Firefox are pushing the "Logout" button to the right.

Should the Header or the navigation tabs handle not having enough space? Maybe by compressing into a dropdown menu instead? Or is the app supposed to detect this and do something different?

Comment (vibdev) [reply modules/_labs/header/react/stories/stories.tsx]: I think user of the component user should decide and pass isCollapsed at the correct time. I think that allows for the most flexibility and makes it so the header doesn't have to do expensive measuring of all different parts anytime there is a change.

**Comment by vibdev** [comment]
closing in favor of https://github.com/Workday/canvas-kit/pull/427

**Pull request closed without merging**